src/pages/RFOX/components/Claim/ClaimConfirm.tsx

```diff
@@ -60,40 +64,65 @@ const backIcon = <ArrowBackIcon />
 const CustomRow: React.FC<RowProps> = props => <Row fontSize='sm' fontWeight='medium' {...props} />
 
 export const ClaimConfirm: FC<Pick<ClaimRouteProps, 'headerComponent'> & ClaimConfirmProps> = ({
-  claimQuote,
+  selectedUnstakingRequest,
   claimTxid,
   setClaimTxid,
 }) => {
   const navigate = useNavigate()
+  const { claimId } = useParams<{ claimId: string }>()
   const translate = useTranslate()
   const wallet = useWallet().state.wallet
+  const dispatch = useAppDispatch()
+
+  const actions = useAppSelector(selectWalletActions)
+
+  const { stakingAssetAccountId, setStakingAssetAccountId } = useRFOXContext()
+
+  useEffect(() => {
+    // Do this as early as possible to ensure the app is properly in sync. Users may have selected a claim from the notification center that's from a
+    // different AccountId than their current one
```

Comment:
Pretty much what the comment says - could end into weird states if not doing that (Txs would still go through, but app would be very confused by the discrepancy and would do things like generating wrong serializedTxIds, meaning status checks would be borked)